src/commands/check.cr

```diff
@@ -15,7 +15,7 @@ module Shards
       end
 
       private def has_dependencies?
-        spec.dependencies.any? || (Shards.with_development? && spec.development_dependencies.any?)
+        !spec.dependencies.empty? || (Shards.with_development? && !spec.development_dependencies.empty?)
```

Comment:
why use `!` and `.empty?` this seems to be less clear.

Reply:
According to Ameba, `!{...}.empty?` is faster than `.any?`, and you can kinda gauge this based on the method implementations:
* [`any?`](https://github.com/crystal-lang/crystal/blob/master/src/enumerable.cr#L111)
* [`empty?`](https://github.com/crystal-lang/crystal/blob/master/src/enumerable.cr#L1496)

Reply:
It's more about the correctness than speed. Since `any?` checks for the first truthy element, it could return `true` even if the enumerable is not empty (given it contains `nil` or `false` values).

Reply:
`any?` vs `!empty?` only makes a difference when items can be falsey. That's not the case here. `spec.depdencies` is of type `Array(Dependency)`, so `any?` has the exact same semantics as `!empty?`.

Now there is still a performance difference, but the implementation of `any?` just needs to be optimized for this case.

IMO an important aspect is code readability and there `any?` clearly wins over `!empty?`.

Reply:
I just checked, in release mode `any?` produces exactly identical code as `empty?` when the item type cannot be falsey (i.e. does not include `Nil`, `Bool` or `Pointer`).
So I don't think this change is justified.

Reply:
Would you be able to provide this code in an issue for Ameba? I think it would be useful to revisit the rule that enforces `empty?` over `any?`. Otherwise, I'm happy to revert this and disable the rule.

Reply:
> IMO an important aspect is code readability and there `any?` clearly wins over `!empty?`.

I disagree. `any?` has a different semantics - checking for truthy elements. Checking whether the collection is empty should be done with call to `empty?` (duh).

Reply:
There's a lot already written on the subject, for instance: https://thoughtbot.com/blog/any-empty

Reply:
I've created a forum post to continue this general discussion: https://forum.crystal-lang.org/t/collections-any-vs-empty/5303

As for this particular change, I'd suggest to drop it and keep the status quo with `any?`. Both versions are semantically equivalent and the current one is more readable, so I see no gain from changing it.

Reply:
Reverted in e7d8a727e0f45ae468a27ef23f0901418c7b5558.

Reply:
> Both versions are semantically equivalent and the current one is more readable

I'm sorry, but please do not rewire words. They're not _semantically_ equivalent - _is (not) empty_ is semantically different thing than: _is there any element (that ... - `any?` takes a block after all)_.

And there's nothing more readable in having to understand the semantics of `any?` to answer this simple question - _empty or not_.